src/ansi-c/expr2c.cpp

```diff
@@ -578,34 +578,39 @@ std::string expr2ct::convert_rec(
   else if(src.id()==ID_vector)
   {
     const vector_typet &vector_type=to_vector_type(src);
+    const auto &size = vector_type.size();
 
-    const mp_integer size_int = numeric_cast_v<mp_integer>(vector_type.size());
-    std::string dest="__gcc_v"+integer2string(size_int);
-
-    std::string tmp=convert(vector_type.subtype());
-
-    if(tmp=="signed char" || tmp=="char")
-      dest+="qi";
-    else if(tmp=="signed short int")
-      dest+="hi";
-    else if(tmp=="signed int")
-      dest+="si";
-    else if(tmp=="signed long long int")
-      dest+="di";
-    else if(tmp=="float")
-      dest+="sf";
-    else if(tmp=="double")
-      dest+="df";
-    else
+    if(size.id() == ID_constant)
```

Comment:
I believe it is an invariant that `vector_type.size().id() == ID_constant` (the key different between arrays and vectors).